iroh-net/src/relay/client.rs

```diff
@@ -1,17 +1,21 @@
 //! based on tailscale/derp/derp_client.go
+use std::marker::PhantomData;
 use std::net::SocketAddr;
+use std::pin::Pin;
 use std::sync::Arc;
+use std::task::{Context, Poll};
 use std::time::Duration;
 
 use anyhow::{anyhow, bail, ensure, Result};
 use bytes::{Bytes, BytesMut};
 use fastwebsockets::{WebSocketError, WebSocketRead, WebSocketWrite};
-use futures_lite::StreamExt;
 use futures_sink::Sink;
 use futures_util::sink::SinkExt;
-use futures_util::TryFutureExt;
-use tokio::io::AsyncWrite;
+use futures_util::stream::{SplitSink, SplitStream};
+use futures_util::{Stream, StreamExt, TryFutureExt};
```

Comment:
we try to avoid these if we can, so please use `futures_lite` as much as possible

Reply:
`futures_lite` re-exposes the same `Stream`, so I can use that, but other than that, there's no alternative to `futures_util::StreamExt`, because I need the `split` function from `futures_util`'s `StreamExt`, `futures_lite::StreamExt` doesn't have that :upside_down_face:

Reply:
yeah, anything related to `Sink` is `futures` only unfortunately